### DISCUSSION TITLE
Question: how to use a flake as a package

### DISCUSSION THREAD

**Discussion by vincentbernat** [nix-community/home-manager#7459] [Q&A]
Hey!

Currently, to use a flake as a package, I do this:

```nix
{
  description = "Home manager flake";
  inputs = {
    nixpkgs.url = "nixpkgs";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    vbeterm = {
      url = "github:vincentbernat/vbeterm";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = inputs:
    let
      pkgs = inputs.nixpkgs.legacyPackages.x86_64-linux;
    in
    {
      homeConfigurations.bernat = inputs.home-manager.lib.homeManagerConfiguration {
        inherit pkgs;
        modules = [ ./home.nix ];
        extraSpecialArgs = { flakes = inputs; };
      };
    };
}
```

Then in `home.nix`, I use:

```nix
{ config, pkgs, lib, flakes, ... }:

{
  programs.home-manager.enable = true;
  home.packages = [
    flakes.vbeterm.packages."x86_64-linux".default
  ];
}
```

This does not seem elegant. Is there a way to not explicitely add the system? Or to shorten this?

**Comment by sullyj3** [comment]
Generally flakes will provide a nixpkgs overlay for this purpose. Eg: 
[neovim-nightly-overlay flake.nix](https://github.com/nix-community/neovim-nightly-overlay/blob/e1ae0a56d37bcb8394f9b9d8ad92119d87cfd19d/flake.nix#L14)
and then I use it in my config like so:
[flake.nix](https://github.com/sullyj3/home-manager-config/blob/d302fd709a3663ca2446a54d2f018531e14688ab/flake.nix#L22)
[common.nix](https://github.com/sullyj3/home-manager-config/blob/d302fd709a3663ca2446a54d2f018531e14688ab/common.nix#L64)
